### PR TITLE
Odsp resolver code refactoring

### PR DESCRIPTION
**Pull request by jatgarg**
1.) Replace telemetry logger to base logger.
2.) Add siteUrl in sharing link token fetcher.
3.) Remove sharing link token fetcher on factory as not required,

### CODE REVIEW

File: packages/drivers/odsp-driver/src/graph.ts [hunk 1]
```diff
@@ -33,7 +33,8 @@ export interface GraphItemLite {
  * @returns Promise which resolves to share link url when successful; otherwise, undefined.
  */
 export async function getShareLink(
-    getShareLinkToken: (options: TokenFetchOptions, isForFileDefaultUrl: boolean) => Promise<string | null>,
+    getShareLinkToken:
+        (options: TokenFetchOptions, isForFileDefaultUrl: boolean, siteUrl: string) => Promise<string | null>,
```
Comment (christiango) [inline packages/drivers/odsp-driver/src/graph.ts:37]: I'm not sure a boolean is the right API here, what if we need to add another one in the future? Would a string union be a better fit here?

Long term, I think we should probably move to the TokenProvider model used in office-web-host and move all the scope business into the driver.

**Review by christiango — APPROVED** [review]

File: packages/drivers/odsp-driver/src/graph.ts (hunk 1 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/graph.ts:37]: Any chance we can rename this arg to something more readable? I can't figure out it's meaning :(

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/graph.ts]: Added enum for this.

File: packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts [hunk 2]
```diff
@@ -50,7 +50,73 @@ export class OdspDriverUrlResolver implements IUrlResolver {
     constructor() { }
 
     public async resolve(request: IRequest): Promise<IOdspResolvedUrl> {
-        return resolveRequest(request);
+        if (request.headers && request.headers[DriverHeader.createNew]) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts:53]: Why do we need this code path? I thought DriverHeader.createNew is no longer used anywhere and should be removed in favor or detached container flow.

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts:53]: DriverHeader.createNew is the detached container path.

**Pull request merged**